Client/Frontend/Browser/OpenSearch.swift

```diff
@@ -36,13 +44,50 @@ class OpenSearchEngine {
     }
 
     /**
+     * Return the arg that we use for searching for this engine
+     * Problem: the search terms may not be a query arg, they may be part of the URL - how to deal with this?
+     **/
+    private func getQueryArgFromTemplate() -> String? {
+        if let queryStartIndex = searchTemplate.rangeOfString("?")?.startIndex.successor() {
+            let queryArgs = searchTemplate.substringFromIndex(queryStartIndex)
+            let queryParts = queryArgs.componentsSeparatedByString("&").map { $0.componentsSeparatedByString("=") }
+            for part in queryParts {
+                if part[1] == SearchTermComponent {
+                    return part[0]
```

Comment:
Unfortunately the placeholder inside the search URL renders it invalid as a URL and therefore you cannot create an NSURLComponents or NSURL directly from the template.

I guess I could replace our placeholder with another one that is unlikely to rock up and test for that.......